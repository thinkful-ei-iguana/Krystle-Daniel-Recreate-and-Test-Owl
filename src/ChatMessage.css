.chat-message {
  display: flex;
  padding-bottom: 0.5rem;
}

.chat-avatar {
  width: 2rem;
}

.chat-avatar img {
  width: inherit;
}

