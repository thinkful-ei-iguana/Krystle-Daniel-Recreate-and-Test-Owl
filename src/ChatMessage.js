import React from 'react';
import './ChatMessage.css';

function ChatMessage (props) {
  let avatar = '';
  let timeName = '';
  let message = '';
  let action = '';
  const part = props.participants;
  const id = props.participantId;
  let time = new Date(props.timestamp).toString();

  if (props.type === 'message') {
    avatar = (
      <div className="chat-avatar">
        <img src={part[id].avatar} alt="avatar"/>
      </div>);
    timeName = (
      <div className="chat-name-time">
        <span>{part[id].name}</span>
        <span>{time}</span>
      </div>);
    message = (
      <div className="chat-message">
        {props.message}
      </div>
    );
  } else if (props.type === 'thumbs-up') {
    action = (
      <div className="chat-action">
        {part[id].name} gave a thumbs up
      </div>
    );
  } else if (props.type === 'raise-hand') {
    action = (
      <div className="chat-action">
        {part[id].name} raised their hand
      </div>
    );
  } else if (props.type === 'thumbs-down') {
    action = (
      <div className="chat-action">
        {part[id].name} gave a thumbs down
      </div>
    );
  } else if (props.type === 'leave-stage') {
    action = (
      <div className="chat-action">
        {part[id].name} left the stage
      </div>
    );
  } else if (props.type === 'join-stage') {
    action = (
      <div className="chat-action">
        {part[id].name} joined the stage
      </div>
    );
  } else if (props.type === 'clap') {
    action =  (
      <div className="chat-action">
        {part[id].name} clapped
      </div>
    );
  } else if (props.type === 'join') {
    action = (
      <div className="chat-action">
        {part[id].name} joined
      </div>
    );
  }
  
  return (
    <div className="chat-message">
      {avatar}
      <div className="chat-message-line">
        {timeName}
        {message}
      </div>
      {action}
    </div>
  );
}

export default ChatMessage;