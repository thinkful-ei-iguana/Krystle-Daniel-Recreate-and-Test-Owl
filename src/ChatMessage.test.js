import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessage from './ChatMessage';

const participants = [
  {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
  {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }
];

describe('ChatMessage component', () => {
    it('renders thumbsup without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='thumbs-up'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders thumbsdown without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='thumbs-down'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders raise-hand without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='raise-hand'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders message without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='message'
          message='Hi'
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders clap without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='clap'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders join without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='join'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders leave without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='leave'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders join-stage without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='join-stage'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders leave-stage without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <ChatMessage 
          participants={participants}
          participantId={0}
          type='leave-stage'
          message=''
          time='1548852646559'
          timestamp='1548852484247'
        />, div);
      ReactDOM.unmountComponentAtNode(div);
    });


  it('renders thumbs-up as expected', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='thumbs-up'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders thumbs-down as expected ', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='thumbs-down'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders raise-hand as expected ', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='raise-hand'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders message as expected', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='message'
        message='Hi'
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders clap as expected', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='clap'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders join as expected ', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='join'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders leave as expected', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='leave'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders join-stage as expected ', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='join-stage'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />).toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders leave-stage as expected', () => {
    const tree = renderer
      .create(<ChatMessage 
        participants={participants}
        participantId={0}
        type='leave-stage'
        message=''
        time='1548852646559'
        timestamp='1548852484247'
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});