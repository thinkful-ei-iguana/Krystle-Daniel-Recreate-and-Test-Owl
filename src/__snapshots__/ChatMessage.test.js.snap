// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage component renders clap as expected 1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     clapped
  </div>
</div>
`;

exports[`ChatMessage component renders join as expected  1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     joined
  </div>
</div>
`;

exports[`ChatMessage component renders join-stage as expected  1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     joined the stage
  </div>
</div>
`;

exports[`ChatMessage component renders leave as expected 1`] = `
<div
  className="chat-message"
>
  
  
  
  
</div>
`;

exports[`ChatMessage component renders leave-stage as expected 1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     left the stage
  </div>
</div>
`;

exports[`ChatMessage component renders message as expected 1`] = `
<div
  className="chat-message"
>
  <div
    className="chat-avatar"
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="chat-name-time"
  >
    <span>
      Koren Templeton
    </span>
    <span>
      Invalid Date
    </span>
  </div>
  <div
    className="chat-message"
  >
    Hi
  </div>
  
</div>
`;

exports[`ChatMessage component renders raise-hand as expected  1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     raised their hand
  </div>
</div>
`;

exports[`ChatMessage component renders thumbs-down as expected  1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     gave a thumbs down
  </div>
</div>
`;

exports[`ChatMessage component renders thumbs-up as expected 1`] = `
<div
  className="chat-message"
>
  
  
  
  <div
    className="chat-action"
  >
    Koren Templeton
     gave a thumbs up
  </div>
</div>
`;
